src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,132 @@
+    void register_existingLogin_notOk() {
+        User userOne = new User();
+        userOne.setAge(19);
+        userOne.setPassword("123456");
+        userOne.setLogin("User");
+        storageDao.add(userOne);
+        User userTwo = new User();
+        userTwo.setAge(19);
+        userTwo.setPassword("123456");
+        userTwo.setLogin("User");
+        invalidDataException
+                = assertThrows(InvalidDataException
+                        .class, () -> registrationService.register(userTwo),
+                String.format("Should throw %s when login is already exists",
+                        EXPECTED_EXCEPTION));
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        User user = new User();
+        user.setAge(19);
+        user.setPassword("123456");
+        user.setLogin(null);
+        invalidDataException
+                = assertThrows(InvalidDataException.class, () -> registrationService.register(user),
+                String.format("Should throw %s when login is null", EXPECTED_EXCEPTION));
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        User user = new User();
+        user.setId(1L);
+        user.setLogin("User");
+        user.setPassword("123456");
+        user.setAge(null);
+        invalidDataException
+                = assertThrows(InvalidDataException.class, () -> registrationService.register(user),
+                String.format("Should throw %s when age is null", EXPECTED_EXCEPTION));
+    }
+
+    @Test
+    void register_under18Age_notOk() {
+        User user = new User();
+        user.setLogin("User");
+        user.setPassword("123456");
+        user.setAge(17);
+        invalidDataException
+                = assertThrows(InvalidDataException
+                        .class, () -> registrationService.register(user),
+                String.format("Should throw %s when age under %d",
+                        EXPECTED_EXCEPTION, MIN_AGE));
+    }
+
+    @Test
+    void register_lessThan6CharsPassword_notOk() {
+        User user = new User();
+        user.setLogin("User");
+        user.setPassword("12345");
+        user.setAge(19);
+        invalidDataException
+                = assertThrows(InvalidDataException.class, () -> registrationService.register(user),
+                String.format("Should throw %s when password less than %d chars",
+                        EXPECTED_EXCEPTION, MIN_PASSWORD_LENGTH));
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        User user = new User();
+        user.setLogin("User");
+        user.setAge(19);
+        try {
+            user.setPassword(null);
+        } catch (NullPointerException e) {
+            invalidDataException
+                    = assertThrows(InvalidDataException
+                            .class, () -> registrationService.register(user),
+                    String.format("Should throw %s when age is null",
+                            EXPECTED_EXCEPTION));
+        }
+    }
+}
```

Comment:
there should be minimum 13 test to check all edgecase scenarios

Reply:
we should test edgecase scenarios: password length 5/6/7  age 17/18/19

Reply:
clear, I`ll add edge case tests scenarios